include/secp256k1_musig.h

```diff
@@ -7,348 +7,152 @@
+ *  guaranteed to be portable between different platforms or versions. It can,
+ *  however, be safely copied/moved. If you need to convert to a format suitable
+ *  for storage, transmission, or comparison, use the corresponding
+ *  serialization and parsing functions.
  */
-typedef struct {
-    uint64_t magic;
-    unsigned char pk_hash[32];
-    unsigned char second_pk[32];
-    int pk_parity;
-    int is_tweaked;
-    unsigned char tweak[32];
-    int internal_key_parity;
-} secp256k1_musig_pre_session;
 
-/** Data structure containing data related to a signing session resulting in a single
- * signature.
- *
- * This structure is not opaque, but it MUST NOT be copied or read or written to it
- * directly. A signer who is online throughout the whole process and can keep this
- * structure in memory can use the provided API functions for a safe standard
- * workflow. See https://blockstream.com/2019/02/18/musig-a-new-multisignature-standard/
- * for more details about the risks associated with serializing or deserializing this
- * structure.
- *
- * Fields:
- *            magic: Set in `musig_session_init` to allow detecting an
- *                   uninitialized object.
- *            round: Current round of the session
- *      pre_session: Auxiliary data created in `pubkey_combine`
- *      combined_pk: MuSig-computed combined xonly public key
- *        n_signers: Number of signers
- *              msg: The 32-byte message (hash) to be signed
- *       is_msg_set: Whether the above message has been set
- *  has_secret_data: Whether this session object has a signers' secret data; if this
- *                   is `false`, it may still be used for verification purposes.
- *           seckey: If `has_secret_data`, the signer's secret key
- *         secnonce: If `has_secret_data`, the signer's secret nonce
- *            nonce: If `has_secret_data`, the signer's public nonce
- * nonce_commitments_hash: If `has_secret_data` and round >= 1, the hash of all
- *                   signers' commitments
- *   combined_nonce: If round >= 2, the summed combined public nonce
- * combined_nonce_parity: If round >= 2, the parity of the Y coordinate of above
- *                   nonce.
+/** Opaque data structure that caches information about public key aggregation.
+ *
+ *  Guaranteed to be 165 bytes in size. No serialization and parsing functions
+ *  (yet).
  */
 typedef struct {
-    uint64_t magic;
-    int round;
-    secp256k1_musig_pre_session pre_session;
-    secp256k1_xonly_pubkey combined_pk;
-    uint32_t n_signers;
-    int is_msg_set;
-    unsigned char msg[32];
-    int has_secret_data;
-    unsigned char seckey[32];
-    unsigned char secnonce[32];
-    secp256k1_xonly_pubkey nonce;
-    int partial_nonce_parity;
-    unsigned char nonce_commitments_hash[32];
-    secp256k1_xonly_pubkey combined_nonce;
-    int combined_nonce_parity;
-} secp256k1_musig_session;
+    unsigned char data[165];
+} secp256k1_musig_keyagg_cache;
```

Comment:
Style nit (feel free to ignore): `keyagg_cache` maybe makes it sound like an optional performance optimization; would `keyagg_state` make the intent more clear?

Reply:
Hm among those I like `keyagg_cache` most. I see the issue with "cache", on the other hand it tells the user that the data is not important and can be thrown away safely. 

Here are two more suggestions:
 - keyagg_context: that fits the other "context" in the library: a thing that keeps state that is necessary for some functions to run but doesn't hold important data and can be recomputed. I really like this a lot now that I think about it. 
 - keyagg_data (that's the most generic but it doesn't have the "cache" issue)

Reply:
I don't have a strong preference! As long as you think the issue with "cache" isn't a big deal then it's probably best to keep that, and not add more noise to the PR by renaming a lot of stuff.

Reply:
Ok, thanks for your input. Let's keep that then.